src/internal_events/adaptive_concurrency.rs

```diff
@@ -13,58 +13,54 @@ pub struct AdaptiveConcurrencyLimit {
     pub past_rtt_deviation: Duration,
 }
 
-impl InternalEvent for AdaptiveConcurrencyLimit {
-    fn emit(self) {
-        trace!(
-            message = "Changed concurrency.",
-            concurrency = %self.concurrency,
-            reached_limit = %self.reached_limit,
-            had_back_pressure = %self.had_back_pressure,
-            current_rtt = ?self.current_rtt,
-            past_rtt = ?self.past_rtt,
-            past_rtt_deviation = ?self.past_rtt_deviation,
-        );
+registered_event! {
+    AdaptiveConcurrencyLimit => {
         // These are histograms, as they may have a number of different
         // values over each reporting interval, and each of those values
         // is valuable for diagnosis.
-        histogram!("adaptive_concurrency_limit", self.concurrency as f64);
-        let reached_limit = self.reached_limit.then_some(1.0).unwrap_or_default();
-        histogram!("adaptive_concurrency_reached_limit", reached_limit);
-        let back_pressure = self.had_back_pressure.then_some(1.0).unwrap_or_default();
-        histogram!("adaptive_concurrency_back_pressure", back_pressure);
-        histogram!("adaptive_concurrency_past_rtt_mean", self.past_rtt);
+        limit: Histogram = register_histogram!("adaptive_concurrency_limit"),
+        reached_limit: Histogram = register_histogram!("adaptive_concurrency_reached_limit"),
+        back_pressure: Histogram = register_histogram!("adaptive_concurrency_back_pressure"),
+        past_rtt_mean: Histogram = register_histogram!("adaptive_concurrency_past_rtt_mean"),
     }
-}
 
-#[derive(Debug)]
-pub struct AdaptiveConcurrencyInFlight {
-    pub in_flight: u64,
+    fn emit(&self, data: AdaptiveConcurrencyLimitData) {
+        self.limit.record(data.concurrency as f64);
+        let reached_limit = data.reached_limit.then_some(1.0).unwrap_or_default();
+        self.reached_limit.record(reached_limit);
+        let back_pressure = data.had_back_pressure.then_some(1.0).unwrap_or_default();
+        self.back_pressure.record(back_pressure);
+        self.past_rtt_mean.record(data.past_rtt);
+        // past_rtt_deviation is unrecorded
```

Comment:
What does this comment mean? Or rather, why is it included if it's not recorded?

Reply:
The existing event did not record the deviation value, and so I didn't add or remove any metrics for it. However, I thought it was worth noting in case we do want to record it.